sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -699,6 +699,194 @@ public void createEndToEndWithNetwork() throws JSONException {
         assertEquals(TEST_FIELD_VALUE, testDocument.test);
     }
 
+    @Test
+    public void createPendingOperationWithoutUpsertHeader() {
+
+        /* If we have one pending operation delete, and the network is on. */
+        final PendingOperation pendingOperation = new PendingOperation(
+                USER_TABLE_NAME,
+                PENDING_OPERATION_CREATE_VALUE,
+                RESOLVED_USER_PARTITION,
+                DOCUMENT_ID,
+                "document",
+                BaseOptions.DEFAULT_EXPIRATION_IN_SECONDS);
+        when(mNetworkStateHelper.isNetworkConnected()).thenReturn(true);
+        when(mLocalDocumentStorage.getPendingOperations(USER_TABLE_NAME)).thenReturn(Collections.singletonList(pendingOperation));
+        Storage.setDataStoreRemoteOperationListener(mDataStoreEventListener);
+
+        /* Setup mock to get valid token from cache. */
+        Calendar expirationDate = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
+        expirationDate.add(Calendar.SECOND, 1000);
+        String tokenResult = Utils.getGson().toJson(new TokenResult()
+                .setPartition(RESOLVED_USER_PARTITION)
+                .setExpirationDate(expirationDate.getTime())
+                .setDbName("db")
+                .setDbAccount("dbAccount")
+                .setDbCollectionName("collection")
+                .setToken(TOKEN));
+        when(SharedPreferencesManager.getString(PREFERENCE_PARTITION_PREFIX + USER)).thenReturn(tokenResult);
+        mockStatic(CosmosDb.class);
+        when(CosmosDb.getUpsertAdditionalHeader()).thenReturn(new HashMap<String, String>() {
+            {
+                put("abc", "bcd");
+            }
+        });
+
+        @SuppressWarnings("unchecked") final ArgumentCaptor<Map<String, String>> headers = ArgumentCaptor.forClass((Class) Map.class);
+        when(CosmosDb.callCosmosDbApi(any(TokenResult.class),
+                anyString(), any(HttpClient.class), anyString(), anyString(), headers.capture(), any(ServiceCallback.class))).then(
+                new Answer<ServiceCall>() {
+
+                    @Override
+                    public ServiceCall answer(InvocationOnMock invocation) {
+                        ((ServiceCallback) invocation.getArguments()[6]).onCallFailed(new HttpException(409, "Conflict happened."));
+                        return mock(ServiceCall.class);
+                    }
+                });
+
+        /* When disable, re-enable to force process pending operations. */
+        Storage.setEnabled(false).get();
```

Comment:
There is no pending operations by definition if you disable. What are we trying to achieve here?

Reply:
just trying to force process he pending operation... directly set true from true I think won't work, but I can try.

Reply:
no, only set it to true directly won't work!

Reply:
This sounds like a hack for test only. In the real SDK implementation, disabling the SDK clears pending operations. Please fix the tests instead of doing this.

Reply:
thanks~